tests/test_helpers.psm1

```diff
@@ -121,16 +130,28 @@ function Get-Port($container, $port=22) {
 }
 
 # run a given command through ssh on the test container.
-function Run-ThruSSH($container, $privateKeyVal, $cmd) {
+function Run-ThruSSH($container, $privateKeyVal, $cmd, $quiet=$false) {
+    $sslLogLevel = 'quiet'
+    if (-not $quiet) {
+        $sslLogLevel = 'verbose'
+        $length = $privateKeyVal.Length
+        Write-Host "Starting Run-ThruSSH with container = $container, privateKeyVal.Length = $length, cmd = $cmd"
+    }
     $SSH_PORT=Get-Port $container 22
     if([System.String]::IsNullOrWhiteSpace($SSH_PORT)) {
         Write-Error "Failed to get SSH port"
         return -1, $null, $null
     } else {
+        if (-not $quiet) {
+            Write-Host "Run-ThruSSH > Get-Port = $SSH_PORT"
+        }
         $TMP_PRIV_KEY_FILE = New-TemporaryFile
         Set-Content -Path $TMP_PRIV_KEY_FILE -Value "$privateKeyVal"
 
-        $exitCode, $stdout, $stderr = Run-Program (Join-Path $PSScriptRoot 'ssh.exe') "-i `"${TMP_PRIV_KEY_FILE}`" -o LogLevel=quiet -o UserKnownHostsFile=NUL -o StrictHostKeyChecking=no -l jenkins localhost -p $SSH_PORT $cmd"
+        $exitCode, $stdout, $stderr = Run-Program (Join-Path $PSScriptRoot 'ssh.exe') "-v -i `"${TMP_PRIV_KEY_FILE}`" -o LogLevel=${sslLogLevel} -o UserKnownHostsFile=NUL -o StrictHostKeyChecking=no -l jenkins localhost -p $SSH_PORT $cmd"

```

Comment:
Weird, the errors on the last builds shows `Connection timed out` as if the container was already removed when `ssh` client is invoked

Reply:
I've added a `docker ps --all` to the helper to check if the container was running, it seems so:

>  Starting Run-ThruSSH with container = pester-jenkins-ssh-agent-jdk11-windowsservercore-ltsc2019, privateKeyVal.Length = 1674, cmd = powershell.exe -NoLogo -C "Write-Host 'f00'"
>  Starting Run-Program with cmd = docker.exe, params = port pester-jenkins-ssh-agent-jdk11-windowsservercore-ltsc2019 22
>  Run-ThruSSH > Get-Port = 50183
>  Starting Run-Program with cmd = docker.exe, params = ps
>  CONTAINER ID   IMAGE                              COMMAND                  CREATED         STATUS         PORTS                   NAMES
>  a2d5fbb9bc29   jdk11-windowsservercore-ltsc2019   "powershell.exe -NoEâ€¦"   6 seconds ago   Up 5 seconds   0.0.0.0:50183->22/tcp   pester-jenkins-ssh-agent-jdk11-windowsservercore-ltsc2019
>  
>  Starting Run-Program with cmd = C:\Jenkins\agent\workspace\ackaging_docker-ssh-agent_PR-295\tests\ssh.exe, params = -v -i "C:\Windows\TEMP\tmpA84A.tmp" -o LogLevel=verbose -o UserKnownHostsFile=NUL -o StrictHostKeyChecking=no -l jenkins localhost -p 50183 powershell.exe -NoLogo -C "Write-Host 'f00'"
>  
>  
>  stdout:
>  
>  
>  stderr:
>  OpenSSH_for_Windows_8.1p1, LibreSSL 2.9.2
>  debug1: Connecting to localhost [::1] port 50183.
>  debug1: connect to address ::1 port 50183: Connection refused
>  debug1: Connecting to localhost [127.0.0.1] port 50183.
>  debug1: connect to address 127.0.0.1 port 50183: Connection timed out
>  ssh: connect to host localhost port 50183: Connection timed out

I'm running out of idea.

Reply:
Found out while working on #388 that waiting a bit before trying to connect with SSH solved the issue:
- https://github.com/jenkinsci/docker-ssh-agent/pull/388#issuecomment-2081254946
- https://github.com/jenkinsci/docker-ssh-agent/pull/391